pkg/JuliaInterface/gap/adapter.gi

```diff
@@ -117,6 +117,101 @@ InstallOtherMethod( \.\:\=,
+##
+BindGlobal( "RandomSourceJulia",
+    rng -> RandomSource( IsRandomSourceJulia, rng ) );
+
+InstallMethod( State,
+    [ "IsRandomSourceJulia and HasJuliaPointer" ],
+    rng -> Julia.Base.copy( JuliaPointer( rng ) ) );
+
+# If the 'JuliaPointer' value is already set then we want to reset
+# an already initialized random source.
+# Then the pair given by 'rng' and its 'JuliaPointer' value may be cached
+# in the Julia session.
+# Thus we cannot replace the 'JuliaPointer' value by a copy.
+# Instead we change the value in place.
+InstallMethod( Init,
+    [ "IsRandomSourceJulia", "IsObject" ],
+    function( rng, seed )
+    ImportJuliaModuleIntoGAP( "Random" );
+    if IsInt( seed ) then
+      # This means a prescribed seed.  (Negative seeds are not supported.)
+      seed:= AbsInt( seed );
+      if HasJuliaPointer( rng ) then
+        Julia.Random.seed\!( JuliaPointer( rng ), GAPToJulia( seed ) );
+      else
+        SetFilterObj( rng, IsAttributeStoringRep );
+        # We did not get a Julia rng, thus we create one by taking
+        # a copy of the default one and then initializing it.
+        SetJuliaPointer( rng,
+            Julia.Random.seed\!( Julia.Base.copy( Julia.Random.default_rng() ),
+                                 GAPToJulia( seed ) ) );
+      fi;
+    elif IsJuliaObject( seed ) and
+         Julia.Base.isa( seed, Julia.Random.AbstractRNG ) then
+      # This means a prescribed state.
+      if HasJuliaPointer( rng ) then
+        Julia.Base.copy\!( JuliaPointer( rng ), seed );
+      else
+        SetFilterObj( rng, IsAttributeStoringRep );
+        # Here we do *not* copy the given Julia rng,
+        # in order to use exactly this rng.
+        SetJuliaPointer( rng, seed );
+      fi;
+    else
+      Error( "<seed> must be a nonnegative integer ",
+             "or a Julia random number generator" );
+    fi;
+    return rng;
+    end );
+
+# The pair given by 'rng' and its 'JuliaPointer' value may be cached
+# in the Julia session.
+# Thus we cannot replace the 'JuliaPointer' value by a copy.
+# Instead we change the value in place.
+InstallMethod( Reset,
+    [ "IsRandomSourceJulia and HasJuliaPointer", "IsObject" ],
+    function( rng, seed )
+    local old;
+
+    old:= State( rng );
+    ImportJuliaModuleIntoGAP( "Random" );
+    if IsInt( seed ) and 0 <= seed then
+      # This means a prescribed seed.
+      Julia.Random.seed\!( JuliaPointer( rng ), GAPToJulia( seed ) );
+    elif IsJuliaObject( seed ) and
+         Julia.Base.isa( seed, Julia.Random.AbstractRNG ) then
+      # This means a prescribed state.
+      Julia.Base.copy\!( JuliaPointer( rng ), seed );
+    else
+      Error( "<seed> must be a nonnegative integer ",
+             "or a Julia random number generator" );
+    fi;
+
+    return old;
+    end );
+
+InstallMethod( Random,
+    [ "IsRandomSourceJulia and HasJuliaPointer", "IsDenseList" ],
+    { rng, list } -> list[ Random( rng, 1, Length( list ) ) ] );
```

Comment:
Is this really necessary? Doesn't the following GAP library method already take care of this?
```gap
InstallMethod(Random, [IsRandomSource, IsList], function(rs, list)
  return list[Random(rs, 1, Length(list))];
end);
```

Reply:
Yes, after today's discussion how to change the setup in GAP, this method does not make sense.

Note that the mentioned  library method is wrong: If the list is not dense then one runs into an error.
In the (intermediate) setup of my pull request for gap-system where non-dense lists were treated "correctly", an additional method for dense list would make sense; but admittedly this method should have been added to GAP itself.